### PR TITLE
🚚(cont.)Bank/Convoy truck convertation to template variable

### PR DESCRIPTION
**Pull request by CalebSerafin**
Original PR By: @igorkis-scrts; `Igorkis-scrts-feature/box-vehicle-template-var`
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Direct usage of `C_Van_01_box_F` classname has been replaced with template variable which will allow to redefine Bank/Convoy missions vehicle for specific modsets.

### Please specify which Issue this PR Resolves.
continues #2562

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
- Spawn the supply mission.
- Confirm that the supply vehicle is created.
- Confirm that the supply vehicle is not persisted after a save/load (As per [(not (_typeVehX in (A3A_faction_reb get "vehiclesCivSupply")))](https://github.com/official-antistasi-community/A3-Antistasi/blob/63537e45ac03f0f4f70b2fbd1fe622916ef87bf7/A3A/addons/core/functions/Save/fn_saveLoop.sqf#L145))

********************************************************
Notes:
@igorkis-scrts, @killerswin2 Because the original branch is now in a read-only repo, this branch will PR directy to Antistasi unstable. 

tl;dr: single vehicleCivSupply -> array vehiclesCivSupply. Merge branch 'feature/box-vehicle-template-var' of https://github.com/igorkis-scrts/A3-Antistasi-Plus into igorkis-scrts-feature/box-vehicle-template-var

### CODE REVIEW

File: A3A/addons/core/functions/Save/fn_saveLoop.sqf [hunk 1]
```diff
@@ -142,7 +142,7 @@ _arrayEst = [];
 	_veh = _x;
 	_typeVehX = typeOf _veh;
 	if ((_veh distance getMarkerPos respawnTeamPlayer < 50) and !(_veh in staticsToSave) and !(_typeVehX in ["ACE_SandbagObject","Land_FoodSacks_01_cargo_brown_F","Land_Pallet_F"])) then {
-		if (((not (_veh isKindOf "StaticWeapon")) and (not (_veh isKindOf "ReammoBox")) and (not (_veh isKindOf "ReammoBox_F")) and (not(_veh isKindOf "Building"))) and (not (_typeVehX == "C_Van_01_box_F")) and (count attachedObjects _veh == 0) and (alive _veh) and ({(alive _x) and (!isPlayer _x)} count crew _veh == 0) and (not(_typeVehX == "WeaponHolderSimulated"))) then {
+		if (((not (_veh isKindOf "StaticWeapon")) and (not (_veh isKindOf "ReammoBox")) and (not (_veh isKindOf "ReammoBox_F")) and (not(_veh isKindOf "Building"))) and (not (_typeVehX in (A3A_faction_reb get "vehiclesCivSupply"))) and (count attachedObjects _veh == 0) and (alive _veh) and ({(alive _x) and (!isPlayer _x)} count crew _veh == 0) and (not(_typeVehX == "WeaponHolderSimulated"))) then {
```
Comment (jaj22) [inline A3A/addons/core/functions/Save/fn_saveLoop.sqf:145]: I'd drop this tbh. Rebels getting a crap free truck occasionally is less of a problem than rebels losing their trucks because this code thinks it's a bank truck.

**Review by jaj22 — CHANGES_REQUESTED** [review]
Otherwise looks fine. Can't think of anything else that's going to trip up.

**Review by jaj22 — APPROVED** [review]
